v6-glm-py/central.py

```diff
@@ -0,0 +1,634 @@
+
+    Returns
+    -------
+    dict
+        The results of the subtask
+    """
+    info("Defining input parameters")
+    input_ = {
+        "method": "compute_local_deviance",
+        "kwargs": {
+            "formula": formula,
+            "family": family,
+            "is_first_iteration": iter_num == 1,
+            "beta_coefficients": beta_estimates,
+            "global_average_outcome_var": global_average_outcome_var,
+        },
+    }
+    if categorical_predictors:
+        input_["kwargs"]["categorical_predictors"] = categorical_predictors
+    if survival_sensor_column:
+        input_["kwargs"]["survival_sensor_column"] = survival_sensor_column
+    if beta_estimates_previous:
+        input_["kwargs"]["beta_coefficients_previous"] = beta_estimates_previous
+
+    # create a subtask for all organizations in the collaboration.
+    info("Creating subtask for all organizations in the collaboration")
+    task = client.task.create(
+        input_=input_,
+        organizations=organizations_to_include,
+        name="Partial deviance subtask",
+        description=f"Subtask to compute partial deviance - iteration {iter_num}",
+    )
+
+    # wait for node to return results of the subtask.
+    info("Waiting for results")
+    results = client.wait_for_results(task_id=task.get("id"))
+    info("Results obtained!")
+
+    # check that each node provided complete results
+    _check_partial_results(results, ["deviance_null", "deviance_old", "deviance_new"])
+
+    return results
+
+
+def _check_partial_results(results: list[dict], required_keys: list[str]) -> None:
+    """
+    Check that each of the partial results contains complete data
+    """
+    for result in results:
+        if result is None:
+            raise AlgorithmExecutionError(
+                "At least one of the nodes returned invalid result. Please check the "
+                "logs."
+            )
+        for key in required_keys:
+            if key not in result:
+                raise AlgorithmExecutionError(
+                    "At least one of the nodes returned incomplete result. Please check"
+                    " the logs."
+                )
+
+
+def _check_input(
+    organizations_to_include: list[int],
+    family: str,
+    formula: str | None,
+    outcome_variable: str | None,
+    predictor_variables: list[str] | None,
+    survival_sensor_column: str | None,
+) -> None:
+    """
+    Check that the input is valid
+
+    Parameters
+    ----------
+    organizations_to_include : list[int]
+        The organizations to include in the computation
+    family : str
+        The family of the GLM
+    formula : str | None
+        The formula to use for the GLM
+    outcome_variable : str | None
+        The name of the outcome variable column
+    predictor_variables : list[str] | None
+        The names of the predictor variable columns
+    survival_sensor_column : str | None
+        The survival_sensor_column value
```

Comment:
:') GPT?

Reply:
I guess so but since I didn't really know myself what it was for when developing the algorithm, I'm not even sure :""-)
anyway changed it now